src/coreclr/src/tools/Common/Compiler/DependencyAnalysis/Relocation.cs

```diff
@@ -8,19 +8,20 @@ namespace ILCompiler.DependencyAnalysis
 {
     public enum RelocType
     {
-        IMAGE_REL_BASED_ABSOLUTE        = 0x00,     // No relocation required
-        IMAGE_REL_BASED_ADDR32NB        = 0x02,     // The 32-bit address without an image base (RVA)
-        IMAGE_REL_BASED_HIGHLOW         = 0x03,     // 32 bit address base
-        IMAGE_REL_BASED_THUMB_MOV32     = 0x07,     // Thumb2: based MOVW/MOVT
-        IMAGE_REL_BASED_DIR64           = 0x0A,     // 64 bit address base
-        IMAGE_REL_BASED_REL32           = 0x10,     // 32-bit relative address from byte following reloc
-        IMAGE_REL_BASED_THUMB_BRANCH24  = 0x13,     // Thumb2: based B, BL
-        IMAGE_REL_BASED_ARM64_BRANCH26  = 0x14,     // Arm64: B, BL
-        IMAGE_REL_BASED_RELPTR32        = 0x7C,     // 32-bit relative address from byte starting reloc
-                                                    // This is a special NGEN-specific relocation type 
-                                                    // for relative pointer (used to make NGen relocation 
-                                                    // section smaller)    
-        IMAGE_REL_SECREL                = 0x80,     // 32 bit offset from base of section containing target
+        IMAGE_REL_BASED_ABSOLUTE             = 0x00,   // No relocation required
+        IMAGE_REL_BASED_ADDR32NB             = 0x02,   // The 32-bit address without an image base (RVA)
+        IMAGE_REL_BASED_HIGHLOW              = 0x03,   // 32 bit address base
+        IMAGE_REL_BASED_THUMB_MOV32          = 0x07,   // Thumb2: based MOVW/MOVT
+        IMAGE_REL_BASED_DIR64                = 0x0A,   // 64 bit address base
+        IMAGE_REL_BASED_REL32                = 0x10,   // 32-bit relative address from byte following reloc
+        IMAGE_REL_BASED_THUMB_BRANCH24       = 0x13,   // Thumb2: based B, BL
+        IMAGE_REL_BASED_THUMB_MOV32_PCREL    = 0x14,   // Thumb2: based MOVW/MOVT
+        IMAGE_REL_BASED_ARM64_BRANCH26       = 0x15,   // Arm64: B, BL
```

Comment:
@AntonLapounov It looks like this relocation isn't handled correctly today in `CorInfoImpl.GetRelocType` . Please fix that there. Also, please adjust the value of this enum to be like the other ARM64 specific relocation types.

Reply:
@davidwrighton What code generates this relocation type?  Apparently none of my tests hit it.

Reply:
@AntonLapounov if we tweak the R2R codegen to generate direct calls/jumps at any point, it will do it. It doesn't now, but it might in the future. 

The two interesting cases are if we come up with a scheme that allows direct calls without going through a R2R entrypoint indirection cell, or when we implement hot/cold splitting of methods. Of the two, I believe that hot/cold splitting is more likely in the .NET 5.0 timeframe, but neither is currently scheduled.